control/shift_decider/src/shift_decider.cpp

```diff
@@ -16,33 +16,42 @@
 
 #include <shift_decider/shift_decider.h>
 
-ShiftDecider::ShiftDecider()
+#include <cstddef>
+#include <functional>
+
+ShiftDecider::ShiftDecider() : Node("shift_decider")
 {
-  pub_shift_cmd_ = pnh_.advertise<autoware_vehicle_msgs::ShiftStamped>("output/shift_cmd", 1, true);
-  sub_control_cmd_ = pnh_.subscribe("input/control_cmd", 1, &ShiftDecider::onControlCmd, this);
-  timer_ = nh_.createTimer(ros::Duration(0.1), &ShiftDecider::onTimer, this);
+  using std::placeholders::_1;
+
+  static constexpr std::size_t queue_size = 1;
+  rclcpp::QoS durable_qos(queue_size);
+  durable_qos.transient_local();
+
+  pub_shift_cmd_ = create_publisher<autoware_vehicle_msgs::msg::ShiftStamped>("output/shift_cmd", durable_qos);
+  sub_control_cmd_ = create_subscription<autoware_control_msgs::msg::ControlCommandStamped>("input/control_cmd", queue_size, std::bind(&ShiftDecider::onControlCmd, this, _1));
+  timer_ = create_wall_timer(std::chrono::duration<double>(0.1), std::bind(&ShiftDecider::onTimer, this));
```

Comment:
We have the same issue with vehicle_cmd_gate regarding timer.
Once investigation is done, we can fix this if needed.

Reply:
@mitsudome-r I added the timer solution as in the `vehicle_cmd_gate`. Please have a final look, hopefully we can then merge this PR